analyzers/src/SonarAnalyzer.CSharp/Rules/EmptyNestedBlock.cs

```diff
@@ -86,19 +43,41 @@ private static bool IsNested(BlockSyntax node)
             SyntaxKind.AnonymousMethodExpression
         };
 
-        private static bool IsEmpty(BlockSyntax node)
-        {
-            return !node.Statements.Any() && !ContainsComment(node);
-        }
+        protected override ILanguageFacade<SyntaxKind> Language => CSharpFacade.Instance;
 
-        private static bool ContainsComment(BlockSyntax node)
+        protected override SyntaxKind[] SyntaxKinds { get; } = new[]
         {
-            return ContainsComment(node.OpenBraceToken.TrailingTrivia) || ContainsComment(node.CloseBraceToken.LeadingTrivia);
-        }
+                SyntaxKind.Block,
+                SyntaxKind.SwitchStatement
+        };
 
-        private static bool ContainsComment(SyntaxTriviaList trivias)
+        protected override IEnumerable<SyntaxNode> EmptyBlocks(SyntaxNode node)
         {
-            return trivias.Any(trivia => trivia.IsKind(SyntaxKind.SingleLineCommentTrivia) || trivia.IsKind(SyntaxKind.MultiLineCommentTrivia));
+            if ((node is SwitchStatementSyntax switchNode && IsEmpty(switchNode))
+                || (node is BlockSyntax blockNode && IsNestedAndEmpty(blockNode)))
+            {
+                return new[] { node };
+            }
+
+            return Enumerable.Empty<SyntaxNode>();
         }
+
+        private static bool IsEmpty(SwitchStatementSyntax node) =>
+            !node.Sections.Any();
+
+        private static bool IsEmpty(BlockSyntax node) =>
+            !node.Statements.Any() && !ContainsComment(node);
+
+        private static bool IsNestedAndEmpty(BlockSyntax node) =>
+            IsNested(node) && IsEmpty(node);
+
+        private static bool IsNested(BlockSyntax node) =>
+            !AllowedContainerKinds.Contains(node.Parent.Kind());
```

Comment:
I wonder if it's more readable to use a switch expression here ([example](https://sharplab.io/#v2:EYLgtghglgdgNAExAagD4AEBMBGAsAKHQGYACLEgYRIG8CT6S6HiSBTGAVzBIBlWAXfqwBONAIJwAQnApwAInACicAGIBfJvU1lSwAPZ6ANiQCSAZzEAzIcIoAKPoJElDAShIBeAHzaGLkmYA7lD8AMYAFoz4fn600TEJjjYAdGJRCRkMeqJJIsmS6ZkZ2bwCKVTeJJYQhmascL4ZAPqeXiT8whysjfRqANwEGvhAA==)) - the values would be here

downside: not sure if it's less readable or not.  (`SyntaxKind.X or SyntaxKind.Y => false, _ => true`) vs `Contains` 
upside: getting rid of a data structure, more explicit

Reply:
Switch expression is unfortunately unavailable for us for now. I changed the `HashSet `to an `Array `and used `IsAnyKind()`.

Reply:
Sorry, I wanted to say that the or pattern is unavailable, not the switch expression.